File: ai_widget/languages/python/agent.py
# Python Integration for Craveva AI Agent
# Install: pip install requests

import requests
import json

class CravevaAgent:
    def __init__(self, agent_id='69ccc35e7d0ece6ff702487b', api_key='cvd_viPEIg-8UsVWKmO7hqfu4LZYxee4AoXE', base_url='https://ai.craveva.com'):
        self.agent_id = agent_id
        self.api_key = api_key
        self.base_url = base_url
        self.api_endpoint = f"{base_url}/api/v1/agents/{agent_id}/chat"
    
    def chat(self, message, user_id=None, company_id=None, outlet_id=None):
        """
        Send a message to the agent and get a response.
        
        Args:
            message (str): The message to send to the agent
            user_id (str, optional): User ID for tracking
            company_id (str, optional): Company ID for data filtering
            outlet_id (str, optional): Outlet ID for data filtering
        
        Returns:
            dict: Response from the agent
        """
        headers = {
            'Content-Type': 'application/json',
        }
        
        if self.api_key:
            headers['Authorization'] = f'Bearer {self.api_key}'
        
        payload = {
            'message': message
        }
        
        if user_id:
            payload['user_id'] = user_id
        if company_id:
            payload['company_id'] = company_id
        if outlet_id:
            payload['outlet_id'] = outlet_id
        
        try:
            response = requests.post(self.api_endpoint, json=payload, headers=headers)
            response.raise_for_status()
            return response.json()
        except requests.exceptions.RequestException as e:
            print(f"Error communicating with agent: {e}")
            return {'error': str(e)}
    
    def stream_chat(self, message, callback=None):
        """
        Stream chat responses (if supported).
        
        Args:
            message (str): The message to send
            callback (callable, optional): Callback function for each chunk
        
        Returns:
            generator: Response chunks
        """
        # Implementation for streaming would go here
        response = self.chat(message)
        if callback:
            callback(response)
        return response


# Usage example:
if __name__ == '__main__':
    agent = CravevaAgent(
        agent_id='69ccc35e7d0ece6ff702487b',
        api_key='cvd_viPEIg-8UsVWKmO7hqfu4LZYxee4AoXE',
        base_url='https://ai.craveva.com'
    )
    
    # Send a message
    response = agent.chat("Hello, how can you help me?")
    print(f"Agent response: {response.get('output', 'No response')}")

File: ai_widget/languages/nodejs/agent.js
// Node.js/Express Integration for Craveva AI Agent
// Install: npm install axios

const axios = require('axios');

class CravevaAgent {
  constructor(agentId = '69ccc35e7d0ece6ff702487b', apiKey = 'cvd_viPEIg-8UsVWKmO7hqfu4LZYxee4AoXE', baseUrl = 'https://ai.craveva.com') {
    this.agentId = agentId;
    this.apiKey = apiKey;
    this.baseUrl = baseUrl;
    this.apiEndpoint = `${baseUrl}/api/v1/agents/${agentId}/chat`;
  }

  /**
   * Send a message to the agent
   * @param {string} message - The message to send
   * @param {Object} options - Optional parameters (userId, companyId, outletId)
   * @returns {Promise<Object>} Response from the agent
   */
  async chat(message, options = {}) {
    const { userId, companyId, outletId } = options;

    const headers = {
      'Content-Type': 'application/json',
    };

    if (this.apiKey) {
      headers['Authorization'] = `Bearer ${this.apiKey}`;
    }

    const payload = {
      message,
    };

    if (userId) payload.user_id = userId;
    if (companyId) payload.company_id = companyId;
    if (outletId) payload.outlet_id = outletId;

    try {
      const response = await axios.post(this.apiEndpoint, payload, {
        headers,
        timeout: 30000,
      });

      return response.data;
    } catch (error) {
      console.error('Error communicating with agent:', error.message);
      return {
        error: error.message,
        output: 'Sorry, I could not process that request.',
      };
    }
  }

  /**
   * Stream chat responses (if supported)
   * @param {string} message - The message to send
   * @param {Function} onChunk - Callback for each chunk
   * @returns {Promise<void>}
   */
  async streamChat(message, onChunk) {
    // Implementation for streaming would go here
    const response = await this.chat(message);
    if (onChunk) {
      onChunk(response);
    }
    return response;
  }
}

// Usage example:
// const agent = new CravevaAgent(
//   '69ccc35e7d0ece6ff702487b',
//   'cvd_viPEIg-8UsVWKmO7hqfu4LZYxee4AoXE',
//   'https://ai.craveva.com'
// );
//
// agent.chat('Hello, how can you help me?')
//   .then(response => {
//     console.log('Agent response:', response.output);
//   })
//   .catch(error => {
//     console.error('Error:', error);
//   });

module.exports = CravevaAgent;
